hazelcast/src/test/java/com/hazelcast/instance/BuildInfoProviderTest.java

```diff
@@ -74,6 +74,22 @@ public void testReadValues() {
 
     }
 
+    @Test
+    public void testCalculateVersion() {
+        assertEquals(-1, BuildInfo.calculateVersion(null));
+        assertEquals(-1, BuildInfo.calculateVersion(""));
+        assertEquals(-1, BuildInfo.calculateVersion("a.3.7.5"));
+        assertEquals(-1, BuildInfo.calculateVersion("3.a.5"));
+        assertEquals(-1, BuildInfo.calculateVersion("3,7.5"));
+        assertEquals(-1, BuildInfo.calculateVersion("3.7,5"));
+        assertEquals(30700, BuildInfo.calculateVersion("3.7-SNAPSHOT"));
+        assertEquals(30702, BuildInfo.calculateVersion("3.7.2"));
+        assertEquals(30702, BuildInfo.calculateVersion("3.7.2-SNAPSHOT"));
+        assertEquals(109902, BuildInfo.calculateVersion("10.99.2-SNAPSHOT"));
+        assertEquals(19930, BuildInfo.calculateVersion("1.99.30"));
+        assertEquals(109930, BuildInfo.calculateVersion("10.99.30-SNAPSHOT"));
```

Comment:
We should have a test for `RC` versions here as well, just to be sure the regex is working properly.